research/condist-fl/README.md

```diff
@@ -9,7 +9,105 @@ Accepted to the 4th Workshop on Distributed, Collaborative, & Federated Learning
 > Developing a generalized segmentation model capable of simultaneously delineating multiple organs and diseases is highly desirable. Federated learning (FL) is a key technology enabling the collaborative development of a model without exchanging training data. However, the limited access to fully annotated training data poses a major challenge to training generalizable models. We propose "ConDistFL", a framework to solve this problem by combining FL with knowledge distillation. Local models can extract the knowledge of unlabeled organs and tumors from partially annotated data from the global model with an adequately designed conditional probability representation. We validate our framework on four distinct partially annotated abdominal CT datasets from the MSD and KiTS19 challenges. The experimental results show that the proposed framework significantly outperforms FedAvg and FedOpt baselines. Moreover, the performance on an external test dataset demonstrates superior generalizability compared to models trained on each dataset separately. Our ablation study suggests that ConDistFL can perform well without frequent aggregation, reducing the communication cost of FL.
 
 ## License
-TBD
+The code in this directory is released under Apache v2 License.
+
+## System Requirements
+
+The sample code is developed and tested on DGX-Station V100 with
+ - NVIDIA V100 32GB * 4
+ - 256GB system memory
+
+Minimam requirements:
+ - 4 CUDA Capable GPUs with > 32GB memory
+ - 96GB system memory
+ - 100GB disk space
+
+## Data Preparation
+Download the liver, pancreas and spleen datasets from the [MSD Challenge](http://medicaldecathlon.com/dataaws/) and the kidney dataset from [KiTS19 Challenge](https://kits19.grand-challenge.org/data/). Once the data is downloaded, use `prepare_data.py` to convert the datasets.
+
+```
+python prepare_data.py --task kidney   --src KiTS19          --dst data/KiTS19
```

Comment:
I'm trying to run the command as such
```
python prepare_data.py --task kidney   --src /data/KiTS19   --dst /tmp/data/KiTS19
python prepare_data.py --task liver    --src /data/Liver    --dst /tmp/data/Liver
python prepare_data.py --task pancreas --src /data/Pancreas --dst /tmp/data/Pancreas
python prepare_data.py --task spleen   --src /data/Spleen   --dst /tmp/data/Spleen
```
but I get
```
root@5037762:/workspace_hroth/Code/nvflare/condist-fl/research/condist-fl# python prepare_data.py --task liver    --src /data/Liver    --dst /tmp/data/Liver
  0%|                                                                                                                                                                                       | 0/131 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "prepare_data.py", line 114, in <module>
    main(args)
  File "prepare_data.py", line 97, in main
    convert_msd_dataset(args.src, args.dst, args.task)
  File "prepare_data.py", line 56, in convert_msd_dataset
    shutil.copy(src_img, dst_img)
  File "/usr/lib/python3.8/shutil.py", line 418, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.8/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/data/Liver/IM_LVR_113.nii.gz'
root@5037762:/workspace_hroth/Code/nvflare/condist-fl/research/condist-fl# python prepare_data.py --task pancreas --src /data/Pancreas --dst /tmp/data/Pancreas
```
I think you need to create the output directory if it doesn't exist. Ideally, the data doesn't have to be in this folder.

Reply:
I have updated the script to support this feature. The default destination path contains a datalist.json for each dataset. The current version will copy the data list to the destination path.